packages/remix-server-runtime/routeModules.ts

```diff
@@ -30,6 +30,24 @@ export interface ActionFunction {
     | AppData;
 }
 
+/**
+ * Route loader function signature.
+ *
+ * This is intended to be used with TypeScript's `satisfies` keyword.
+ *
+ * @example
+ *
+ * ```ts
+ * export const loader = (({ request }) => {
+ *   const url = new URL(request.url);
+ *   return new Response("hello from " + url.pathname);
+ * }) satisfies LoaderFunction;
+ * ```
+ */
+export interface LoaderFunction {
+  (args: DataFunctionArgs): Promise<Response> | Response;
```

Comment:
We may need to iterate on the return type a little.

Eg. do we want to allow loader functions to return `null | Response`?

Reply:
This already exists later on this this file?

Reply:
Oh! I don't have this, maybe because I'm on an experimental remix build (deferred channel)

Reply:
ah that makes sense :)

Reply:
@tom-sherman if you can remove changes to this file, the other doc changes you've made look good to me!